inventory/src/components/ItemList/BlueprintItemTable.js

```diff
@@ -0,0 +1,263 @@
+      }
+      return item;
+    });
+  };
+  const handleInputChange = (id, quantity, totalQty) => {
+    if (quantity !== '') {
+      quantity = parseInt(quantity);
+    }
+    if (quantity === '' || (0 <= quantity && quantity <= totalQty)) {
+      setItemsToDisplay((prevItems) =>
+        prevItems.map((item) => {
+          if (item.id === id) {
+            return { ...item, selectedQty: quantity };
+          }
+          return item;
+        }),
+      );
+      items.forEach((item) => {
+        if (item.id === id) {
+          item.selectedQty = quantity;
+          updateSelectedItems(item.id, item.name, item.selectedQty);
+          return item;
+        }
+      });
+    } else {
+      // error handling
+    }
+  };
+
+  const handleOnBlur = (id) => {
+    setItemsToDisplay((prevItems) =>
+      prevItems.map((item) => {
+        if (item.id === id && item.selectedQty === '') {
+          return { ...item, selectedQty: 0 };
+        } else if (item.id === id && item.selectedQty > item.quantity) {
+          return { ...item, selectedQty: item.quantity };
+        }
+        return item;
+      }),
+    );
+    items.forEach((item) => {
+      if (item.id === id && item.selectedQty === '') {
+        item.selectedQty = 0;
+      } else if (item.id === id && item.selectedQty > item.quantity) {
+        item.selectedQty = item.quantity;
+      }
+    });
+  };
+
+  return (
+    <Box
+      sx={{
+        width: {
+          xs: '95%',
+          sm: '90%',
+          md: '70%',
+        },
+      }}
+    >
+      <Stack spacing={3} alignItems={'center'}>
+        <Box
+          sx={{
+            width: 1,
+            display: 'flex',
+            alignItems: 'center',
+            marginBottom: 2,
+          }}
+        >
+          <TextField
+            label='Search'
+            value={searchTerm}
+            onChange={(e) => setSearchTerm(e.target.value)}
+            sx={{ width: 1 }}
+          />
+          <Select
+            inputProps={{ 'data-testid': 'type-select' }}
+            value={currentType}
+            onChange={(e) => setCurrentType(e.target.value)}
+          >
+            {types.map((t) => (
+              <MenuItem key={t} value={t}>
+                {t}
+              </MenuItem>
+            ))}
+          </Select>
+        </Box>
+        <TableContainer component={Paper}>
+          <Table>
+            <TableHead>
+              <TableRow>
+                <TableCell>Item Name</TableCell>
+                <TableCell align={'center'}>Unit</TableCell>
+                <TableCell align={'center'}>Total Qty</TableCell>
+                <TableCell align={'center'}>
+                  Qty To Add (Capped Between 0 & Total Qty)
+                </TableCell>
+                <TableCell></TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {itemsToDisplay.slice(startIndex, endIndex).map((item) => (
```

Comment:
would be good if we can include an `<EmptyMessge />` when there are no items to display that matches the user's query